src/Backend/Haskell.idr

```diff
@@ -40,10 +98,11 @@ data Haskell : Type where
   ||| and a number of constructors, each wrapping a Haskell type.
   ADT     : Decl -> Vect n (Name, HsType) -> Haskell
 
--- Haskell type variables start with lowercase letters.
-private
-freshEnv : Env n
-freshEnv = freshEnv "x"
+  ||| A function definition is a declared name, a type, and a list of
+  ||| clauses of the form ((arg1, ..., argk), rhs).
+  FunDef : Name -> HsType -> List (List HsTerm, HsTerm) -> Haskell
```

Comment:
Shouldn't the list of `HsTerm` all be of the same length, wouldn't `Vect n HsTerm` be more appropriate?

Reply:
In principle yes, but it is sometimes not so straightforward to compute the `n` (e.g. we skip trivial `T1` arguments to constructors), so for now I think it is easiest to keep this invariant unchecked.

Reply:
Alright